src/routes/transactions/transactions.controller.spec.ts

```diff
@@ -612,6 +612,109 @@ describe('Transactions Controller (Unit)', () => {
         });
     });
   });
+
+  describe('GET queued transactions by Safe', () => {
+    it('should get a transactions queue with labels and conflict headers', async () => {
+      const chainId = faker.random.numeric();
+      const safeAddress = faker.finance.ethereumAddress();
+      const chainResponse = new ChainBuilder().withChainId(chainId).build();
+      mockNetworkService.get.mockImplementation((url) => {
+        const getChainUrl = `${safeConfigApiUrl}/api/v1/chains/${chainId}`;
+        const getMultisigTransactionsUrl = `${chainResponse.transactionService}/api/v1/safes/${safeAddress}/multisig-transactions/`;
+        const getSafeUrl = `${chainResponse.transactionService}/api/v1/safes/${safeAddress}`;
+        const getContractUrlPattern = `${chainResponse.transactionService}/api/v1/contracts/`;
+        if (url === getChainUrl) {
+          return Promise.resolve({ data: chainResponse });
+        }
+        if (url === getMultisigTransactionsUrl) {
+          return Promise.resolve({
+            data: getJsonResource(
```

Comment:
Yes, using factories here seems more appropriate, thanks, I will change it. 👍🏻

Reply:
@fmrsabino I pushed [06716c5](https://github.com/5afe/safe-client-gateway-nest/pull/215/commits/06716c571741022b78adee62a90c5c2ae6b1de3d) using some builders just as sneak peak. I see the utility of using builders but I also feel we are hardcoding a little too much data in each test and making them a bit harder to read. Wdyt? Maybe building the objects in a separate file could be a intermediate solution?

Reply:
I've set the PR as a draft because I feel the change in [06716c5](https://github.com/5afe/safe-client-gateway-nest/pull/215/commits/06716c571741022b78adee62a90c5c2ae6b1de3d) could be improved (extracting common data, etc.) but I'm not sure this time-consuming change adds a lot of value right now, and I think we have to be conscious of time at this point. Should we remove the usage of `getJsonResource` completely from this file? This would imply also composing the expected responses using builders, and also refactoring all the tests in the file. Of course, we can do it in a separate PR, but I have doubts about doing this work at this point since it's time-consuming. What do you think? @fmrsabino @moisses89  🤔